src/main/scala/monix/nio/tcp/AsyncTcpClientConsumer.scala

```diff
@@ -50,6 +59,7 @@ class AsyncTcpClientConsumer(host: String, port: Int) extends Consumer[Array[Byt
 
       override def onNext(elem: Array[Byte]): Future[Ack] = {
         val promise = Promise[Ack]()
+        connectedSignal.await(timeoutInSeconds, TimeUnit.SECONDS)
```

Comment:
Blocking threads isn't cool :-) 

As a suggestion your `connectedSignal` could be a `Future[Unit]`. And as an optimization you could work with its `.value.isEmpty` to know whether you should `flatMap` on it or not. Or you could work with a `Task` that's `memoized`, along with `runTry`.